src/index.rs

```diff
@@ -913,6 +913,45 @@ impl Index {
       parent,
     }))
   }
+  pub(crate) fn rune_json(&self, rune: Rune) -> Result<Option<RuneJson>> {
+    let rtx = self.database.begin_read()?;
+
+    let Some(id) = rtx
+      .open_table(RUNE_TO_RUNE_ID)?
+      .get(rune.0)?
+      .map(|guard| guard.value())
+    else {
+      return Ok(None);
+    };
+
+    let entry = RuneEntry::load(
+      rtx
+          .open_table(RUNE_ID_TO_RUNE_ENTRY)?
+          .get(id)?
+          .unwrap()
+          .value(),
+    );
+
+    let parent = InscriptionId {
+      txid: entry.etching,
+      index: 0,
+    };
+
+    let parent = rtx
+        .open_table(INSCRIPTION_ID_TO_SEQUENCE_NUMBER)?
+        .get(&parent.store())?
+        .is_some()
+        .then_some(parent);
+
+    let rune_id = RuneId::load(id);
+    Ok(Some(RuneJson {
+      entry,
+      height: rune_id.height,
+      index: rune_id.index,
+      id: RuneId::load(id),
+      parent,
+    }))
+  }
```

Comment:
This function is almost identical to rune_html and combined with the suggestion below you can construct the Html and Json in the same function.

Reply:
thanks for your suggestion